### PR TITLE
GREP15: More detailed info about pybind11 implementation

### PR DESCRIPTION
**Pull request by mormj**

### CODE REVIEW

File: grep-0015-remove-swig.md [hunk 1]
```diff
@@ -27,18 +29,156 @@ anyway.
...
 
-**To be written**. GREP14 (blocktool) would be a help.
+### Block Binding Code Structure
+
+In each gr module, create a `python/bindings` directory
+
+`python/bindings`
+-  `python_bindings.cpp` <-- c++ code to build up the pybind11 module block by block
+   -  alternatively this can be broken up into a per-block cpp file
+-  `generated/` <-- folder to contain automatically generated binding code with [blockname]_python.hpp
```
Comment (mormj) [inline grep-0015-remove-swig.md:50]: On second thought, I think this is an unnecessary distinction, especially in-tree.  Let's just put all the binding files in the bindings directory for simplicity

File: grep-0015-remove-swig.md [hunk 2]
```diff
@@ -27,18 +29,156 @@ anyway.
...
+{
+public:
+    typedef std::shared_ptr<char_to_float> sptr;
+    static sptr make(size_t vlen = 1, float scale = 1.0);
+    virtual float scale() const = 0;
+    virtual void set_scale(float scale) = 0;
+};
+```
+
+In order to bind this, we write/generate `char_to_float_python.hpp`:
```
Comment (mormj) [inline grep-0015-remove-swig.md:66]: Let's just make this char_to_float_python.cc, and treat as a separate compilation unit

File: grep-0015-remove-swig.md [hunk 3]
```diff
@@ -1,21 +1,23 @@
 # GREP 0015 -- Replace SWIG with something else
 
 - Original Author: Martin Braun <martin@gnuradio.org>
-- Champion: Martin Braun <martin@gnuradio.org> (Looking for a new champion!)
+- Champion: Martin Braun <martin@gnuradio.org> / Josh Morman <mormjb@gmail.com>
```
Comment (mbr0wn) [inline grep-0015-remove-swig.md:4]: You can have this line if you want :)

File: grep-0015-remove-swig.md [hunk 4]
```diff
@@ -1,21 +1,23 @@
...
 
 ## Abstract
 
 **Note: This is a very raw GREP. It's a discussion we've been meaning to have
 for a while. This GREP will stay here for as long as this repo will exist, even
 if the out come of the discussion is to not replace SWIG.**
 
-SWIG is a huge dependency. We might be able to do better. One option is to use
-Boost.Python, since Boost is already a dependency. Potential gains:
+SWIG is a huge dependency. We might be able to do better.  The leading candidate is pybind11, which is a header only library to produce Python bindings from c++. Potential gains:
```
Comment (mbr0wn) [inline grep-0015-remove-swig.md:17]:
```suggestion
SWIG is a huge dependency. We might be able to do better.  The leading candidate is pybind11, which is a header only library to produce Python bindings from C++. Potential gains:
```

File: grep-0015-remove-swig.md [hunk 5]
```diff
@@ -27,18 +29,156 @@ anyway.
 ## Copyright / License
 
 This GREP is licensed as CC-BY-ND.
-Copyright 2018 The GNU Radio Foundation.
+Copyright 2020 Free Software Foundation??
```
Comment (mbr0wn) [inline grep-0015-remove-swig.md:32]: Just put our names here.

File: grep-0015-remove-swig.md [hunk 6]
```diff
@@ -27,18 +29,156 @@ anyway.
...
 
 ## Motivation
 
 A lot of us have cursed SWIG programming. If we can get rid of it... well,
 wouldn't that be great. But maybe we can't. Let's find out.
```
Comment (mbr0wn) [inline grep-0015-remove-swig.md:37]: I guess we've found out that we can remove it:

```suggestion
A lot of us have cursed SWIG programming. That alone is a good motivation for removing it.
```

**Review by mbr0wn — COMMENTED** [review]
Wow I reviewd this months ago and forgot to click "submit".

**Comment by mbr0wn** [comment]
Merged manually.

**Pull request closed without merging**